BedrockCore.cpp

```diff
@@ -107,11 +107,14 @@ void BedrockCore::prePeekCommand(unique_ptr<BedrockCommand>& command) {
             SALERT("Command " << command->request.methodLine << " timed out after " << e.time()/1000 << "ms.");
         }
         STHROW("555 Timeout prePeeking command");
+        command->complete = true;
```

Comment:
I think the line above this is wrong, we shouldn't re-throw the timeout, we need to handle the timeout here, the same as for peek. Peek catches this *inside* a separate try/catch block and then does the `STHROW` so that when we get to this point, it's already an SException.

I don't think the throw above gets caught anywhere. Try getting a command to timeout in prePeek, I bet it crashes.

Reply:
You must have started your review right moments before I pushed. Haha.